### ISSUE TITLE
new fields in the schema (added to the postgres schema) begin to be recognized only after failed and restarting the `clickhouse-sink-connector`

### ISSUE THREAD

**Issue by salisbury-espinosa** [Altinity/clickhouse-sink-connector#904]
so we have to stop the `clickhouse-sink-connector` with an error  and start `clickhouse-sink-connector`.
"hot reload" is not available and causes an error during runtime

```bash
sink  | java.sql.SQLException: Missing value for parameter #74 [test2 String]
sink  | 	at com.clickhouse.jdbc.SqlExceptionUtils.clientError(SqlExceptionUtils.java:73) ~[clickhouse-kafka-sink-connector-0.0.9.jar:?]
sink  | 	at com.clickhouse.jdbc.internal.InputBasedPreparedStatement.addBatch(InputBasedPreparedStatement.java:333) ~[clickhouse-kafka-sink-connector-0.0.9.jar:?]
```

"just stop clickhouse-sink-connector, upgrade schema, and start clickhouse-sink-connector" - doesn't work.
it only works with an error - maybe something is being cached or there is some kind of race condition

Example upgrade schema in postgresql:
```sql
ALTER TABLE test ADD COLUMN test2 character varying(50) NOT NULL DEFAULT 'yiii';
```

pay attention to `NOT NULL`

at the same time, the schema in the clickhouse itself is updated and a new column appears !

**Comment by minguyen9988** [comment]
Verified fixed at develop head `d41888aa`.

Both halves of what you hit are addressed.

1. The stale cached column map (the `Missing value for parameter #74 [test2 String]` symptom) is fixed by generation-counter cache invalidation. `CacheInvalidationManager` versions each table; DDL bumps it (`DebeziumChangeEventCapture.java:800`) and any cached writer whose recorded version is stale is discarded and rebuilt:

`ClickHouseBatchWriter.java:290-309`
```java
long currentVersion = CacheInvalidationManager.getInstance().getVersion(fullyQualifiedTableName);
DbWriter writer = this.topicToDbWriterMap.get(topicName);
if (writer != null) {
    if (writer.getCacheInvalidationVersion() == currentVersion) { return writer; }
    this.topicToDbWriterMap.remove(topicName);
```

2. The ordering race is closed by `drainBeforeDDL()` (`DebeziumChangeEventCapture.java:1249`), which hands buffered pre-DDL rows to the consumers before the DDL is applied — so rows written just before the ALTER are not bound against the post-ALTER column map.

The restart is no longer needed for the new column to be recognized. Note the `NOT NULL` detail in your report was incidental; the defect was the cached `DbWriter` column map.

Closing as fixed.
